// @ampdev/common/build/events/base-listener.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Listener = void 0;
class Listener {
    constructor(client) {
        this.acWait = 5 * 1000;
        this.client = client;
    }
    subscriptionOption() {
        return this.client
            .subscriptionOptions()
            .setManualAckMode(true)
            .setDeliverAllAvailable()
            .setAckWait(this.acWait)
            .setDurableName(this.qGroupName);
    }
    listen() {
        const subscription = this.client.subscribe(this.subject, this.qGroupName, this.subscriptionOption());
        subscription.on('message', (msg) => {
            console.log(`Message Received: ${this.subject} / ${this.qGroupName}`);
            const parsedData = this.paresMesssage(msg);
            this.onMessage(parsedData, msg);
        });
    }
    paresMesssage(msg) {
        const data = msg.getData();
        return typeof data === 'string'
            ? JSON.parse(data)
            : JSON.parse(data.toString('utf8'));
    }
}
exports.Listener = Listener;
